presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -168,19 +169,28 @@ else if (config.getPinGlueClientToCurrentRegion()) {
             }
         }
 
+        AWSCredentialsProvider credentialsProvider = createAwsCredentialsProvider(config);
+        asyncGlueClientBuilder.setCredentials(credentialsProvider);
+        return asyncGlueClientBuilder.build();
+    }
+
+    private static AWSCredentialsProvider createAwsCredentialsProvider(GlueHiveMetastoreConfig config)
+    {
+        AWSCredentialsProvider credentialsProvider;
+
         if (config.getAwsAccessKey().isPresent() && config.getAwsSecretKey().isPresent()) {
-            AWSCredentialsProvider credentialsProvider = new AWSStaticCredentialsProvider(
-                    new BasicAWSCredentials(config.getAwsAccessKey().get(), config.getAwsSecretKey().get()));
-            asyncGlueClientBuilder.setCredentials(credentialsProvider);
+            credentialsProvider = new AWSStaticCredentialsProvider(
+                new BasicAWSCredentials(config.getAwsAccessKey().get(), config.getAwsSecretKey().get()));
         }
         else if (config.getIamRole().isPresent()) {
-            AWSCredentialsProvider credentialsProvider = new STSAssumeRoleSessionCredentialsProvider
-                    .Builder(config.getIamRole().get(), "presto-session")
-                    .build();
-            asyncGlueClientBuilder.setCredentials(credentialsProvider);
+            credentialsProvider = new STSAssumeRoleSessionCredentialsProvider
+                .Builder(config.getIamRole().get(), "presto-session")
+                .build();
         }
-
-        return asyncGlueClientBuilder.build();
+        else {
+            credentialsProvider = new DefaultAWSCredentialsProviderChain();
```

Comment:
You should probably call `DefaultAWSCredentialsProviderChain.getInstance()`

(then you should rename the method to `getAwsCredentialsProvider`, the same way it was in s3 code)

Reply:
Will do